rust/kcl-lib/src/std/args.rs

```diff
@@ -560,24 +560,23 @@ impl Args {
         Ok(())
     }
 
-    pub(crate) fn make_user_val_from_point(&self, p: [TyF64; 2]) -> Result<KclValue, KclError> {
+    pub(crate) fn make_kcl_val_from_point(&self, p: [f64; 2], ty: NumericType) -> Result<KclValue, KclError> {
```

Comment:
Rather than all the changes to the segment functions and having the docs regression, this function could keep the original signature and adjust the units of the second number to match the first. I think there is already a utility function to do that (and of course this is a pretty rare case, in practice the two coords always have the same units).

Reply:
It's asking for bugs to make a general helper function to have such weird logic. Rust types are a proof of the shape of the data. The path store the information that x and y share the same units. We shouldn't throw that away. As a workaround until we make smarter docs generation, I contained duplication logic in the segment module where it requires the bad function signature. The `inner_*` functions are designed to only be called from one place.